src/Exception/NotFoundException.php

```diff
@@ -2,11 +2,13 @@
 
 namespace Http\Discovery\Exception;
 
+use Http\Discovery\Exception;
```

Comment:
Exception\Exception sounds weird, also, the inheritance is also reflected in the naming this way, which seems to be logical to me.